osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuHitObject.cs

```diff
@@ -10,9 +10,23 @@ namespace osu.Game.Rulesets.Osu.Objects.Drawables
 {
     public class DrawableOsuHitObject : DrawableHitObject<OsuHitObject>
     {
+        /// <summary>
+        /// The number of milliseconds before <see cref="Rulesets.Objects.HitObject.StartTime"/> we should start fading in.
+        /// <para>This should be calculated in the future.</para>
+        /// </summary>
         public const float TIME_PREEMPT = 600;
         public const float TIME_FADEIN = 400;
-        public const float TIME_FADEOUT = 500;
+
+        public bool Hidden;

```

Comment:
no

Reply:
Is it the name or the bool itself?
Should I go back to the 4 different ones I had before?

Reply:
can you list the four you are requiring? generally drawable hitobjects should not be aware of hidden at all (the mod should be doing all the moving)

Reply:
I had
`ShowApproachCircle` and `PlayHitAnimation` for DrawableHitCircle,
`FadeOutGradually` for DrawableSlider and
`HideSpinnerDetails` for DrawableSpinner

Reply:
Let's go with those for now

Reply:
Please make sure to add xmldoc on each explaining what they do, will make it easier to follow for future adventurers

Reply:
Going through it only 2 may be sufficient though. 🤔 
ShowApproachCircle and AllowEarlyFadeOut.
The latter would already describe what I need for all 3 types anyways. Also I need something in DrawableOsuHitObject too that allows me to use a different calculation for FadeIn (determined by a bool named `hidden` in stable).
Technically the description would be correct as the HitObjects do fade out earlier. Partially or completely.

Reply:
the different calculation should be done by the mod itself, not the drawable implementation